app/domain/authentication/authn_azure/azure_resource.rb

```diff
@@ -0,0 +1,13 @@
+module Authentication
+  module AuthnAzure
+
+    class AzureResource
+      attr_reader :resource_type, :resource_name
+
+      def initialize(resource_type:, resource_name:)
+        @resource_type = resource_type
```

Comment:
type & name are similar in their meaning. IMO it will be easier to diffentiate with 

```
resource-name ("subscription-id") // this can also be resource-type btw. which ever one you see fit
resource-value (" xms_mirid.subscription")
```

Reply:
yeah originally it was resource name and value as you did. I changed it to align with k8s who already had the use of the resource type. I think i can have it the other way around and align k8s to have name and value.

Reply:
According to [k8s docs](https://kubernetes.io/docs/reference/kubectl/overview/#resource-types), it seems that "resource type" is more appropriate.

Reply:
or maybe Resource Kind

Reply:
I don't mind if it is `resource type` / `kind` as long as the second is `resource-value` b/c when I think `value`, I think key -> value which is what you are trying to get build

Reply:
makes sense, thanks

Reply:
The class is already named `AzureResource`, so repeating that in the member names is called stutter.

`type` and `value` read smoother.